src/readii/negative_controls.py

```diff
@@ -372,11 +365,16 @@ def applyNegativeControl(baseImage: Union[Image, ndarray],
             return makeRandomImage(baseImage, randomSeed)
         elif negativeControlType == "randomized_sampled":
             return makeRandomSampleFromDistributionImage(baseImage, randomSeed)
-    elif negativeControlRegion == "roi":
+    
+    assert roiMask is not None, \
+        f"ROI mask is None. Must pass ROI mask to negative control function for {negativeControlType} negative control."
+    
+    if negativeControlRegion == "roi":
         return negativeControlROIOnly(baseImage, roiMask, negativeControlType, randomSeed)
     elif negativeControlRegion == "non_roi":
         return negativeControlNonROIOnly(baseImage, roiMask, negativeControlType, randomSeed)
-
+    else:
+        raise ValueError(f"Invalid negative control region: {negativeControlRegion}. Must be one of ['full', 'roi', 'non_roi']")
```

Comment:
Is this not caught by the if statement above?

Reply:
ah youre right. Since the function declares that it will return either a Image or ndarray, without handling the `else` it can potentially return None. Ill fix that.